modules/connatixBidAdapter.js

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable no-template-curly-in-string */
+import {
+  deepAccess,
+  isFn,
+  logError,
+  isArray
+} from '../src/utils.js';
+
+import {
+  registerBidder
+} from '../src/adapters/bidderFactory.js';
+import {
+  BANNER,
+} from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'connatix';
+const AD_URL = 'https://capi.connatix.com/rtb/hba';
+const DEFAULT_MAX_TTL = '3600';
+const DEFAULT_CURRENCY = 'USD';
+
+/*
+   * Get the bid floor value from the bid object, either using the getFloor function or by accessing the 'params.bidfloor' property.
+   * If the bid floor cannot be determined, return 0 as a fallback value.
+   */
+export function getBidFloor(bid) {
+  if (!isFn(bid.getFloor)) {
+    return deepAccess(bid, 'params.bidfloor', 0);
+  }
+
+  try {
+    const bidFloor = bid.getFloor({
+      currency: DEFAULT_CURRENCY,
+      mediaType: '*',
+      size: '*',
+    });
+    return bidFloor.floor;
+  } catch (err) {
+    logError(err);
+    return 0;
+  }
+}
+
+/*
+   * Wrap the provided bid, playerId, customerId, and scriptId in an HTML string
+   * that includes the Connatix player script and related data.
+   * This HTML string will be used as the ad content.
+   */
+export function wrapAd(lineItems, requestId, playerId, customerId, isInApp) {
+  var settings = {
+    advertising: {
+      standaloneLineItems: lineItems
+    }
+  };
+  var scriptSrc = isInApp ? `//cd.connatix.com/connatix.player.omid.js?cid=${customerId}` : `//cd.connatix.com/connatix.player.js?cid=${customerId}`;
```

Comment:
Hi @rares-mihai-preda,

Can you please tell us about the script that you are loading here. What's it's supposed to do and why do you need it?

Reply:
@Fawke If the source is from mobile applications, we utilize the first script; whereas, if it is from browsers, we employ the second script.

Reply:
The current state of isInApp is always false, making the check unnecessary so I removed the ternary operator.

The script includes Connatix's player code and its monetization, meaning that the bid is injected via external API to the Connatix player for rendering purposes.

Reply:
@patmmccann I'm not sure if this violates rule#2 of the [Global Module Rules](https://docs.prebid.org/dev-docs/module-rules.html). (Modules must not load outside code files unless those libraries are approved and open source).

Can you please have a look? Thanks.

Reply:
@Fawke @patmmccann Allow me to add additional details here. This script is our ads rendering core. It will be added to the bid creative after the auction ends and it doesn't affect the Prebid.js core/bundle. It will be loaded as part of the creative in the creative HTML string when our creative will be rendered. It will never be added from the bidder's code prior to our creative getting rendered.

Reply:
why not just specify a renderer like everyone else? https://github.com/prebid/Prebid.js/blob/bcf080d381d3f650a03e1808e89d5e517600b151/modules/rubiconBidAdapter.js#L833

Reply:
@patmmccann I updated and removed this part. We no longer needed it.